engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/query/SplitFieldsByUnderlyingService.kt

```diff
@@ -0,0 +1,57 @@
+package graphql.nadel.enginekt.transform.query
+
+import graphql.nadel.OperationKind
+import graphql.nadel.Service
+import graphql.nadel.enginekt.schema.FieldCoordinatesToUnderlyingServiceMapping
+import graphql.nadel.enginekt.util.copyWithChildren
+import graphql.nadel.enginekt.util.makeFieldCoordinates
+import graphql.normalized.NormalizedField
+import graphql.normalized.NormalizedQuery
+import graphql.schema.GraphQLSchema
+
+class SplitFieldsByUnderlyingService {
+
+    fun execute(query: NormalizedQuery, schema: GraphQLSchema, fieldInfos: FieldCoordinatesToUnderlyingServiceMapping, operationKind: OperationKind): List<FieldAndService> {
+        return query.topLevelFields
+            .flatMap { topLevelField ->
+                val isNamespacedField = topLevelField.getOneFieldDefinition(schema).getDirective("namespaced") != null
```

Comment:
Can we add a `NadelNamespaced` object that does some of this logic?

```
object NadelNamespaced {
  private val namespacedDirectiveName = "namespaced"

  fun isNamespacedField(schema: Schema, field: GraphQLFieldDefinition): Boolean {
    return …
  }
}
```

Reply:
put the `object NadelNamespacedFields` in the same class, do you suggest extracting it in a separate file?

Reply:
Yea separate file would be great.